File: test3.cpp
#include<iostream>
#include<string>

int panduan(std::string s)
{
  int a = 0, b = 0, c = 0, q = 0, z = 0, h = 0;

  for(int i = 0; i < s.size(); i++)
  {
    if(s[i] == 'A')
    {
      if(a == 0 && b == 0)
        q++;
      else if(a == 1 && b == 0)
        z++;
      else if(a == 1 && b == 1)
        h++;
    }
    else if(s[i] == 'P')
      a++;
    else if(s[i] == 'T')
      b++;
    else
    {
      c = 0;
      return c;
    }
  }

  if(h == q * z && z >= 1 && a == 1 && b == 1)
    c = 1;

  return c;
}

int main()
{
  int a, b;
  std::string s;

  std::cin >> a;

  int *t = new int[a];

  for(int i = 0; i < a; i++)
  {
    std::cin >> s;
    t[i] = panduan(s);
  }

  for(int j = 0; j < a; j++)
  {
    if(t[j] == 1)
      std::cout << "YES" << std::endl;
    else
      std::cout << "NO" << std::endl;
  }
  system("pause");
}

File: test3.py
def panduan(s):
  a = b = c = q = z = h = 0

  for i in range(len(s)):
    if(s[i] == 'A'):
      if(a == 0 and b == 0):
        q = q + 1
      elif(a == 1 and b == 0):
        z = z + 1
      elif(a == 1 and b == 1):
        h = h + 1
    elif(s[i] == 'P'):
      a = a + 1
    elif(s[i] == 'T'):
      b = b + 1
    else:
      c = 0
      return c

  if(h == q * z and z >= 1 and a == 1 and b == 1):
    c = 1

  return c

a = eval(input())
t = []

for i in range(a):
  s = input()
  t.append(panduan(s))

for i in range(a):
  if(t[i] == 1):
    print("YES")
  else:
    print("NO")
